org/uacalc/alg/Malcev.java

```diff
@@ -732,6 +732,84 @@ public static Term fixedKEdgeTerm(SmallAlgebra alg, int k, ProgressReport report
     return null;
   }
 
+  /** 
+   * Alexandr Kazda's implementation (based on bits of previous UACalc code)
+   * of the local to global test for quasi weak near uninimity operation of
+   * arity <code>arity</code>. See the paper A. Kazda "Deciding the existence
+   * of quasi weak near unanimity terms in finite algebras"
+   *
+   * 
+   * @param testalg  an algebra 
+   * @param arity arity (at least 2)
+   * @return Boolean
+   */
+  public static boolean fixedKQwnu(SmallAlgebra testalg, int arity, ProgressReport report) {
+    if (arity < 2) throw new IllegalArgumentException("arity must be at least 2");
+    if (report != null) report.addStartLine("Using the local to global method to test for quasi WNU of arity " + arity + ". See the paper A. Kazda \"Deciding the existence of quasi weak near unanimity terms in finite algebras,\" 2020."); 
+    int power = arity;
+    BigProductAlgebra alg = new BigProductAlgebra(testalg, power);
+    final int size = testalg.cardinality();
+    //IntArray genvector = new IntArray(arity);
+    int a;
+    int b;
+    int i;
+    int j;
+    int[][] G= new int[arity][arity];
+    int[] F=new int[arity];
+    boolean found;
+    for (a=0;a<size;a++) {
+	    Arrays.fill(F,a);
+		    for(i=0;i<arity;i++)
+			    System.arraycopy(F,0,G[i],0,arity);//Fill the whole G with a's
+	    for(b=0;b<size;b++) {
+    		    List<IntArray> gens = new ArrayList<IntArray>(arity);
+		    if (a==b)
```

Comment:
Doesn't this end up being the same as
```
for (b=0; b<a; b++) {...
```
?

Reply:
I only want to skip a=b since then the test always succeeds. The case b>a can happen.